app/services/shopify_import/variants/data_parsers/base_data.rb

```diff
@@ -0,0 +1,49 @@
+module ShopifyImport
+  module Variants
+    module DataParsers
+      class BaseData
+        def initialize(shopify_variant, spree_product)
+          @shopify_variant = shopify_variant
+          @spree_product = spree_product
+        end
+
+        def variant_attributes
+          {
+            sku: @shopify_variant.sku,
+            price: @shopify_variant.price,
+            weight: @shopify_variant.grams,
+            position: @shopify_variant.position,
+            product_id: @spree_product.id
+          }
+        end
+
+        def option_value_ids
+          %w[option1 option2 option3].map do |option_name|
+            next unless (option_value = @shopify_variant.send(option_name))
+
+            Spree::OptionValue.find_by!(
```

Comment:
Doesn't this raise an error when there's no such OptionValue in Spree store? Should it instead create the option value?